File: C_Matlab/interpolation_analysis/t3_205727183.c
#include "nr.h"
#include "nrutil.h"
#include "math.h"
#include "stdio.h"

float function(float *f, float x){
*f = x*sin(2*x)*fabs(x-0.75);
}

int main(){

float f;
float df;
int i;
float x;

float *xa = vector(1,11);
float *ya = vector(1,11);

for(i = 1;i<12;i++){
xa[i] = (i-1)*0.15;
function(&f, xa[i]);
ya[i] = f;
}

i = 0;
FILE *fout;
fout = fopen("poli10.txt","w");
for(i = 0;i<100;i++){
x = i*0.0151515;
polint(xa,ya,10,x,&f,&df);
fprintf(fout,"%f %f %f\n",x,f,df);
}

float *y2 = vector(1,11);

FILE *fid;
fid = fopen("spl.txt","w");
spline(xa,ya,10,0,0,y2);
i = 0;

for(i = 0;i<100;i++){
x = i*0.0151515;
splint(xa,ya,y2,10,x,&f);
fprintf(fid,"%f %f\n",x,f);
}



}

File: C_Matlab/interpolation_analysis/t3_205727183.m
s = load('poli10.txt');
x = zeros(100,1);
for m = 1:100
    x(m) = s(m,1);
end
f1 = zeros(100,1);
for m = 1:100
    f1(m) = s(m,2);
end
df1 = zeros(100,1);
for m = 1:100
    df1(m) = s(m,3);
end
figure
plot(x,(x.*sin(2.*x)).*abs(x-0.75),'b');
hold on;
errorbar(x,f1,df1,'r');

s2 = load('spl.txt');
x2 = zeros(100,1);
for m = 1:100
    x2(m) = s2(m,1);
end
f2 = zeros(100,1);
for m = 1:100
    f1(m) = s2(m,2);
end
figure
plot(x,(x.*sin(2.*x)).*abs(x-0.75),'b');
hold on;
plot(x,f1,'r');
